src/DotVVM.Framework/Controls/UpdateProgress.cs

```diff
@@ -21,6 +36,11 @@ protected override void AddAttributesToRender(IHtmlWriter writer, IDotvvmRequest
         {
             writer.AddKnockoutDataBind("dotvvm-UpdateProgress-Visible", "true");
 
+            if (Delay != 0)
```

Comment:
What if the delay is a negative number? https://stackoverflow.com/questions/8430966/is-calling-settimeout-with-a-negative-delay-ok